fe/parameterize_batch.py

```diff
@@ -0,0 +1,387 @@
+
+            all_du_dls = []
+
+            sim = simulation.Simulation(
+                guest_mol,
+                host_pdb,
+                'insertion',
+                step_sizes,
+                cas,
+                lambda_schedule,
+                current_off_params
+                )
+
+            init_conf = guest_mol.GetConformer(0)
+            init_conf = np.array(init_conf.GetPositions(), dtype=np.float64)
+            init_conf = init_conf/10 # convert to md_units
+            conf_com = com(init_conf)
+            init_mol = Chem.Mol(guest_mol)
+            num_conformers = args.num_conformers
+
+            # generate a set of gas phase conformers using the RDKit
+            guest_mol.RemoveAllConformers()
+            AllChem.EmbedMultipleConfs(guest_mol, num_conformers, randomSeed=2020)
+            np.random.seed(2020)
+
+            for conf_idx in range(num_conformers):
+                conformer = guest_mol.GetConformer(conf_idx)
+                guest_conf = np.array(conformer.GetPositions(), dtype=np.float64)
+                guest_conf = guest_conf/10 # convert to md_units
+                rot_matrix = special_ortho_group.rvs(3).astype(dtype=np.float64)
+                guest_conf = np.matmul(guest_conf, rot_matrix)*10
+
+                for atom_idx, pos in enumerate(guest_conf):
+                    conformer.SetAtomPosition(atom_idx, (float(pos[0]), float(pos[1]), float(pos[2])))
+
+            all_args = []
+            for conf_idx in range(num_conformers):
+
+                conformer = guest_mol.GetConformer(conf_idx)
+                guest_conf = np.array(conformer.GetPositions(), dtype=np.float64)
+                guest_conf = guest_conf/10 # convert to md_units
+                guest_conf = recenter(guest_conf, conf_com)
+
+                x0 = np.concatenate([host_conf, guest_conf])       # combined geometry
+
+                combined_pdb = Chem.CombineMols(Chem.MolFromPDBFile(host_pdb_file, removeHs=False), init_mol)
+                combined_pdb_str = StringIO(Chem.MolToPDBBlock(combined_pdb))
+                out_file = os.path.join(args.frames_dir, "epoch_"+str(epoch)+"_insertion_deletion_"+host_name+"_conf_"+str(conf_idx)+".pdb")
+                writer = PDBWriter(combined_pdb_str, out_file)
+
+                # set this to None if we don't care about visualization
+                all_args.append([x0, writer, conf_idx % num_gpus, precision, None])
+
+            all_du_dls = pool.map(sim.run_forward_multi, all_args)
+            all_du_dls = np.array(all_du_dls)
+
+            loss_grad_fn = jax.grad(error_fn, argnums=(0,))
+          
+            for du_dls in all_du_dls:
+                fwd = du_dls[:T//2]
+                bkwd = du_dls[T//2:]
+
+            error = error_fn(all_du_dls, T, lambda_schedule, true_dG)
+
+            print("---EPOCH", epoch, "---- LOSS of one ligand in training ----", error)
+
+            error_grad = loss_grad_fn(all_du_dls, T, lambda_schedule, true_dG)
```

Comment:
not sure what true_dG here is pointing to since you don't use the list of true_dGs anywhere